File: server/src/api/user/service/user.service.ts
import type { User } from '@/api/user/model/user.model';
import { UserRepository } from '@/api/user/repo/user.repository';
import { BaseService } from '@/core/api/base.service'

export class UserService extends BaseService<User, UserRepository> {

  constructor() {
    super(new UserRepository());
  }

}

export const userService = new UserService();

File: backend/app/api/services/user_service.py
from uuid import UUID

from app.api.models.user import User
from app.api.repos.user_repository import UserRepository
from app.api.services.base_service import BaseService

class UserService(BaseService[User]):
    def __init__(self, session):
        super().__init__(User, session, UserRepository)

    def update_avatar(self, user_id: UUID, avatar_url: str):
        user = self.get(user_id)
        return self.repo.update(user, {"avatar": avatar_url})
